charts/hivemq-platform/templates/hivemq-configuration.yml

```diff
@@ -162,87 +159,155 @@ data:
         {{- end }}
       </data-hub>
       {{- end }}
-      {{- if .Values.hivemqRestrictions }}
+      {{- $hasRestrictionsConfig := include "hivemq-platform.has-hivemq-restrictions-config" (dict "hivemqRestrictions" .Values.hivemqRestrictions) }}
+      {{- if $hasRestrictionsConfig }}
+      {{- $restrictionsConfig := .Values.hivemqRestrictions }}
       <restrictions>
-        <max-connections>{{ .Values.hivemqRestrictions.maxConnections | default -1 | int64 }}</max-connections>
-        <incoming-bandwidth-throttling>{{ .Values.hivemqRestrictions.incomingBandwidthThrottling | default 0 | int64 }}</incoming-bandwidth-throttling>
-        <no-connect-idle-timeout>{{ .Values.hivemqRestrictions.noConnectIdleTimeout | default 1000 | int64 }}</no-connect-idle-timeout>
-        <max-client-id-length>{{ .Values.hivemqRestrictions.maxClientIdLength | default 65535 }}</max-client-id-length>
+        {{- if (hasKey $restrictionsConfig "maxConnections") }}
+        <max-connections>{{ $restrictionsConfig.maxConnections | int64 }}</max-connections>
```

Comment:
I like it very much, that we don't have to duplicate the default values here anymore!